src/SprykerEco/Zed/UnzerApi/Business/Api/Request/Builder/UnzerApiRequestBuilder.php

```diff
@@ -42,8 +42,38 @@ public function __construct(
      */
     public function buildRequestPayload(UnzerApiRequestTransfer $unzerApiRequestTransfer): string
     {
-        $requestData = $this->unzerApiRequestConverter->convertUnzerApiRequestTransferToArray($unzerApiRequestTransfer);
+        $requestData = $this->unzerApiRequestConverter
+            ->convertUnzerApiRequestTransferToArray($unzerApiRequestTransfer);
 
-        return (string)$this->utilEncodingService->encodeJson($requestData);
+        $this->formatAmounts($requestData);
+
+        return (string)$this->utilEncodingService->encodeJson($requestData, $requestData === [] ? JSON_FORCE_OBJECT : null);
+    }
+
+    /**
+     * A workaround to avoid json_encode precision issue
+     * Formats float amountGross value into string of expected format
+     *
+     * @param array $requestData
+     *
+     * @return void
+     */
+    protected function formatAmounts(array &$requestData): void
+    {
+        if (isset($requestData['canceledBasket']['items'])) {
+            foreach ($requestData['canceledBasket']['items'] as &$item) {
```

Comment:
It is recommended to use `unset` when you work with array items passed by reference.